File: iraf/unix/boot/spp/rpp/ratlibf/ctoi.f
      integer function ctoi(in, i)
      integer in (100)
      integer i
      integer d
      external index
      integer index
      integer digits(11)
      data digits (1) /48/, digits (2) /49/, digits (3) /50/, digits (4)
     * /51/, digits (5) /52/, digits (6) /53/, digits (7) /54/, digits (
     *8) /55/, digits (9) /56/, digits (10) /57/, digits (11) /-2/
23000 if (.not.(in (i) .eq. 32 .or. in (i) .eq. 9))goto 23001
      i = i + 1
      goto 23000
23001 continue
      ctoi = 0
23002 if (.not.(in (i) .ne. -2))goto 23004
      d = index (digits, in (i))
      if (.not.(d .eq. 0))goto 23005
      goto 23004
23005 continue
      ctoi = 10 * ctoi + d - 1
23003 i = i + 1
      goto 23002
23004 continue
      return
      end

File: unix/boot/spp/rpp/ratlibr/ctoi.r
include	defs

# ctoi - convert string at in(i) to integer, increment i

   integer function ctoi(in, i)
   character in (ARB)
   integer i

   integer d
   external index
   integer index

   # string digits "0123456789"
   character digits(11)
   data digits (1) /DIG0/,
      digits (2) /DIG1/,
      digits (3) /DIG2/,
      digits (4) /DIG3/,
      digits (5) /DIG4/,
      digits (6) /DIG5/,
      digits (7) /DIG6/,
      digits (8) /DIG7/,
      digits (9) /DIG8/,
      digits (10) /DIG9/,
      digits (11) /EOS/

   while (in (i) == BLANK | in (i) == TAB)
      i = i + 1
   for (ctoi = 0; in (i) != EOS; i = i + 1) {
      d = index (digits, in (i))
      if (d == 0)      # non-digit
	 break
      ctoi = 10 * ctoi + d - 1
      }

   return
   end
